subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/PluginDeclaration.java

```diff
@@ -30,7 +30,9 @@
  * @see org.gradle.plugin.devel.plugins.JavaGradlePluginPlugin
  * @since 2.14
  */
-public class PluginDeclaration implements Named, Serializable {
+public class PluginDeclaration implements Named,
+    // TODO: Shouldn't be serializable, remove the interface in Gradle 8.0.
+    Serializable {
```

Comment:
How are we going to find this in 8.0? Shall we create the PR already?

Reply:
Yes, I think it makes sense to create a PR for 8.0 after this one has been merged.

Reply:
See https://github.com/gradle/gradle/pull/20097

Reply:
@lptr @wolfs Can you add a deprecation warning for this? e.g. via one/all of:
```java
private void readObject(ObjectInputStream ois) throws IOException, ClassNotFoundException {
    // emit warning about serialization
    ois.defaultReadObject();
}
private void writeObject(ObjectOutputStream oos) throws IOException, ClassNotFoundException {
    // emit warning about serialization
    oos.defaultWriteObject();
}
private Object readResolve()  {
    // emit warning about serialization
    return this;
}
```

Reply:
That is a good idea! I tried it here: https://github.com/gradle/gradle/pull/20125

There is currently a problem that the deprecation warning also is emitted when using configuration caching, so I am not sure I can include the deprecation.

Reply:
Continued here: https://github.com/gradle/gradle/pull/20125/files#r822429036